app/validators/ssh_key_validator.rb

```diff
@@ -0,0 +1,15 @@
+class SshKeyValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    if value && !valid_ssh_public_key?(value)
+      record.errors[attribute] << _('is invalid')
+    end
+  end
+
+  private
+
+  def valid_ssh_public_key?(key)
+    SSHKey.valid_ssh_public_key?(key)
+  rescue SSHKey::PublicKeyError
+    false
```

Comment:
`false` is the intended and correct return value. See my other comment why I chose the other method to return `nil`.